cli/flox-watchdog/src/logger.rs

```diff
@@ -15,26 +15,19 @@ const KEEP_WATCHDOG_DAYS: u64 = 3;
 const KEEP_SERVICES_LAST: usize = 5;
 
 /// Initializes a logger that persists logs to an optional file in addition to `stderr`
-pub(crate) fn init_logger(file_path: &Option<PathBuf>) -> Result<(), anyhow::Error> {
+pub(crate) fn init_logger(
+    logs_dir: &Option<PathBuf>,
+    log_file_prefix: &str,
+) -> Result<(), anyhow::Error> {
     let stderr_layer = tracing_subscriber::fmt::layer()
         .with_writer(std::io::stderr)
         .with_filter(EnvFilter::from_default_env());
-    let file_layer = if let Some(path) = file_path {
-        let path = if path.is_relative() {
-            std::env::current_dir()?.join(path)
-        } else {
-            path.clone()
-        };
-        let file = OpenOptions::new()
-            .create(true)
-            .truncate(true)
-            .write(true)
-            .open(&path)
-            .with_context(|| format!("failed to open log file {}", path.display()))?;
+    let file_layer = if let Some(dir_path) = logs_dir {
+        let appender = tracing_appender::rolling::daily(dir_path, log_file_prefix);
```

Comment:
question blocking: so does this just split up logs into 1 file for every day? That could still grow infinitely because we only `spawn_gc_logs` once at startup, so I don't think that really accomplishes the goal. Unless `tracing_appender` cleans stuff up

Reply:
Ah, I misread because the heartbeat one loops and that one doesn't. I'll just add a loop in the gc logs one too.